systemverilog-plugin/UhdmAst.cc

```diff
@@ -2173,13 +2174,64 @@ void UhdmAst::process_typespec_member()
     add_multirange_wire(current_node, packed_ranges, unpacked_ranges);
 }
 
-void UhdmAst::process_enum_typespec()
+void UhdmAst::process_enum_typespec(const UHDM::BaseClass *object)
 {
     // BaseTypespec specifies underlying type of the enum.
     // The BaseTypespec has at most one explicit packed dimension (range).
     // When base type is not specified in SystemVerilog code, it is assumed to be an int.
     // Type of enum items (constants) is the same as the enum type.
     current_node = make_ast_node(AST::AST_ENUM);
+
+    const auto *enum_object = object->Cast<const UHDM::enum_typespec *>();
+    const auto *typespec = enum_object->Base_typespec();
+
+    if (typespec && typespec->UhdmType() == UHDM::uhdmlogic_typespec) {
+        // If it's a logic_typespec, try to reduce expressions inside of it.
+
+        const auto *logic_typespec_obj = enum_object->Base_typespec()->Cast<const UHDM::logic_typespec *>();
+        std::vector<UHDM::range *> ranges;
+        // Check if ranges exist.
+        if (logic_typespec_obj->Ranges()) {
+            ranges = logic_typespec_obj->Ranges()[0];
```

Comment:
This is a little bit confusing, why do we need to select ``[0]`` element? Please add comment what is internal structure of ``logic_typespec_obj->Ranges()``.

Reply:
Now I realise it could be confusing indeed, as I used it to dereference a pointer.
I've modified it to use the asterisk dereference operator, and expanded the comment explaining that `Ranges()` returns a pointer.